starknet_py/net/models/abi/definition.py

```diff
@@ -0,0 +1,180 @@
+from __future__ import annotations
+
+from collections import OrderedDict, defaultdict
+from dataclasses import dataclass
+from typing import Dict, List, TypeVar, Optional, cast
+
+from marshmallow import EXCLUDE
+
+from starknet_py.cairo.type_parser import TypeParser
+from starknet_py.cairo.data_types import StructType, CairoType
+from starknet_py.net.models.abi.schemas import ContractAbiEntrySchema
+from starknet_py.net.models.abi.shape import (
+    StructDict,
+    FunctionDict,
+    TypedMemberDict,
+    EventDict,
+    STRUCT_ENTRY,
+    FUNCTION_ENTRY,
+    EVENT_ENTRY,
+    CONSTRUCTOR_ENTRY,
+    L1_HANDLER_ENTRY,
+    StructMemberDict,
+)
+
+
+class AbiDecodingError(ValueError):
+    pass
+
+
+T = TypeVar("T", bound=Dict)
```

Comment:
Can we use more descriptive name to suggest it is in fact something like a dict?